yesod-form/Yesod/Form/Fields.hs

```diff
@@ -768,6 +822,26 @@ selectFieldHelper outside onOpt inside opts' = Field
             x -> case olReadExternal opts x of
                     Nothing -> Left $ SomeMessage $ MsgInvalidEntry x
                     Just y -> Right $ Just y
+-- |
+-- A helper function for constucting 'selectField's with option groups. You may want to use this when you define your custom 'selectField's or 'radioField's.
+selectFieldHelperGrouped
+        :: (Eq a, RenderMessage site FormMessage)
+        => (Text -> Text -> [(Text, Text)] -> WidgetFor site () -> WidgetFor site ()) -- ^ Outermost part of the field
+        -> (Text -> Text -> Bool -> WidgetFor site ()) -- ^ An option for None if the field is optional
+        -> (Text -> Text -> [(Text, Text)] -> Text -> Bool -> Text -> WidgetFor site ()) -- ^ Other options
+        -> (Text -> WidgetFor site ()) -- ^ Group headers placed inbetween options
+        -> HandlerFor site (OptionList a)
+        -> Field (HandlerFor site) a
+selectFieldHelperGrouped outside onOpt inside group opts' = do
+    let optsFlat = fmap flattenOptionList opts'
+    selectFieldHelper outside onOpt inside optsFlat -- {
+    --   fieldView = \theId name attrs val isReq -> do
```

Comment:
@snoyberg Here I'm trying change only part of the `Field` (i.e. `fieldView`). It should be possible right?